add tag-and-probe physics model

cecilecaillol opened the pull request:
New physics model for tag and probe measurements. The POI is the multiplicative factor for the signal in the pass category, while the number of signal events in the fail region is simultaneously modified by an anticorrelated factor such that the sum of signal events in pass and fail regions remains constant.

nucleosynthesis commented:
This would make a very nice addition, however, can the same not be
accomplished by using ...
https://twiki.cern.ch/twiki/bin/view/CMS/HiggsWG/SWGuideNonStandardCombineUses#Rate_Parameters

by adding to cards the expression produced in this line ....
https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/242/files#diff-bbec2e0a4630ad4a1b8365aa38091114R28
?

If we merge this in, a simple example datacard would  be nice and could be
added to the above Twiki for other users.

On Tue, Sep 29, 2015 at 6:00 PM, cecilecaillol <notifications@github.com>
wrote:

> New physics model for tag and probe measurements. The POI is the
> multiplicative factor for the signal in the pass category, while the number
> of signal events in the fail region is simultaneously modified by an
> anticorrelated factor such that the sum of signal events in pass and fail
> regions remains constant.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/242
> Commit Summary
>
>    - add TnP physics model
>
> File Changes
>
>    - *A* python/TagAndProbeModel.py
>    <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/242/files#diff-0>
>    (43)
>
> Patch Links:
>
>    -
>    https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/242.patch
>    -
>    https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/242.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/242>.
>

gpetruc commented on python/TagAndProbeModel.py:
```diff
@@ -0,0 +1,43 @@
+                        return "fail_scale"
+                else:
+                        poi=1
+                        return int(poi)
```
can't this be just `return 1` ? (same for the one below)

gpetruc commented on python/TagAndProbeModel.py:
```diff
@@ -0,0 +1,43 @@
+    def setPhysicsOptions(self,physOptions):
+        for po in physOptions:
+            if po.startswith("anticorrelation="):
+                self.anticorrelation = float(po.replace("anticorrelation=",""))
```
was this supposed to be used somewhere?

gpetruc commented on python/TagAndProbeModel.py:
```diff
@@ -0,0 +1,43 @@
+                    exp_pass=self.DC.exp[b][p]
+                 if re.search("fail",b):
+                    exp_fail=self.DC.exp[b][p]
+        self.modelBuilder.factory_('expr::fail_scale("(%f+%f-(%f*@0))/%f", SF)'%(exp_pass,exp_fail,exp_pass,exp_fail))
```
I'm not totally convinced of the math here, in case one has additional systematics which may cause the yield to change wrt `DC.exp[b][p]` (plus, it fails if you fit two bins at once)

A simpler implementation could be to fill the signal expectation with the full yield both in the passing and in the failing category (i.e. as if the efficiency were 100% and 0% respectively) and then return eff and (1-eff) in the two. That way if you had e.g. a normalization systematic scaling the signal yield up or down by 10% you would still get the right answer.

gpetruc commented:
Hello,

Thanks for the model, 

I've marked a two places where the code may be cleaned up a bit (so that new people reading it may not get confused). Maybe you could make those changes, commit and push again to your github branch to update this PR.

For the third comment on the normalization, there may also be a way to get the expected signal yield after systematic in order to make `fail_scale` to be correct even in the presence of systematics if the proposal of filling the signal and background with the full yield seems inconvenient (or even problematic,  e.g. if passing signal and failing signal have different shapes), but it would need a bit more thinking.

cecilecaillol commented:
Thank you for the comments! I have committed a new version of the code to my github branch, where I have removed the useless "physics option" part, and simplified the "return 1". I have not changed anything yet concerning the third comment on the normalization.

nucleosynthesis commented:
@cecilecaillol I see there hasn't been any activity on the 3rd comment but it seems we can go ahead and merge, leaving the correction to take the full normalisation after systematics as a future improvement. Thoughts?

cecilecaillol commented:
Thanks!
